server/src/main/java/org/spine3/server/command/CommandBus.java

```diff
@@ -236,43 +237,58 @@ public void unregister(CommandHandler handler) {
      * @param responseObserver the observer to return the result of the call
      */
     public void post(Command command, StreamObserver<Response> responseObserver) {
-        final CommandClass commandClass = CommandClass.of(command);
+        checkNotNull(command);
+        checkNotNull(responseObserver);
+        checkArgument(isNotDefault(command));
+
+        final CommandEnvelope commandEnvelope = new CommandEnvelope(command);
+        final CommandClass commandClass = commandEnvelope.getCommandClass();
+
+        final Optional<CommandEndpoint> commandEndpoint = getEndpoint(commandClass);
 
         // If the command is not supported, return as error.
-        if (!isSupportedCommand(commandClass)) {
+        if (!commandEndpoint.isPresent()) {
             handleUnsupported(command, responseObserver);
             return;
         }
+
         if (!filter.handleValidation(command, responseObserver)) {
             return;
         }
+
         if (isScheduled(command)) {
             scheduleAndStore(command, responseObserver);
             return;
         }
+
         if (isMultitenant) {
```

Comment:
Should we set the current tenant _before_ handling the command?

I would move the tenant update onto the spot before `if ... { handleUnsupported (..) }`, since even handling the unsupported commands should generally depend on the target tenant.